### PR TITLE
fix OdnoklassnikiServiceImpl signature

### PR DESCRIPTION
**Pull request by aliksz**
query parameters must be sorted in lexicographic order before hashing

### CODE REVIEW

**Comment by kullfar** [comment]
Is it the new Odnoklassniki requirement?
The code is working well for me

**Comment by kullfar** [comment]
Travis said

All checks have failed
1 errored check

File: scribejava-apis/src/main/java/com/github/scribejava/apis/service/OdnoklassnikiServiceImpl.java [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import com.github.scribejava.core.model.OAuthConfig;
 import com.github.scribejava.core.oauth.OAuth20Service;
+import org.apache.commons.codec.CharEncoding;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.apache.commons.codec.digest.DigestUtils.md5Hex;
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/service/OdnoklassnikiServiceImpl.java]: do not resort the imports pls. It's unneeded diff

File: scribejava-apis/src/test/java/com/github/scribejava/apis/serviceTest/OdnoklassnikiServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import com.github.scribejava.apis.OdnoklassnikiApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.*;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Created by alex on 22.08.16.
```
Comment (kullfar) [inline scribejava-apis/src/test/java/com/github/scribejava/apis/serviceTest/OdnoklassnikiServiceTest.java]: git blame is enough

File: scribejava-apis/src/test/java/com/github/scribejava/apis/serviceTest/OdnoklassnikiServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Test
+    public void shouldThrowExceptionIfBaseStringIsNull() {
+        OAuth2AccessToken accessToken = new OAuth2AccessToken("d3iwa.403gvrs194740652m1k4w2a503k3c");
+        OAuthRequest request = new OAuthRequest(Verb.GET, URL, service);
+        service.signRequest(accessToken, request);
+
+        //session_secret_key = md5Hex(accessToken.getAccessToken() + service.getConfig().getApiSecret()) == 129ea5bb368dfb321dc90f63da01b742
+        //значения_параметров  == application_key=AAAAAAAAAAAAAAAAfields=uid%2Cfirst_name%2Clast_name%2Cpic_2format=jsonmethod=friends.get
+        //sig == MD5(URLDecoder.decode(значения_параметров, CharEncoding.UTF_8) + session_secret_key) == 96127f5ca29a8351399e94bbd284ab16
```
Comment (kullfar) [inline scribejava-apis/src/test/java/com/github/scribejava/apis/serviceTest/OdnoklassnikiServiceTest.java]: do not leave comments in the final code

**Comment by kullfar** [comment]
Am I right, that you have errrors while running
https://github.com/scribejava/scribejava/blob/master/scribejava-apis/src/test/java/com/github/scribejava/apis/examples/OdnoklassnikiExample.java
And errors are gone with your PullRequest?

**Comment by aliksz** [comment]
Not, i use  this library for production code.
I have incorrect signature for my requests. 
According to http://new.apiok.ru/dev/methods/  rule for sign requests as follows: 

**Расчет подписи запроса**

при отсутствии значения session_secret_key:
для вызова без сессии считаем session_secret_key = application_secret_key
для вызова в сессии session_secret_key = MD5(access_token + application_secret_key), переводим значение в нижний регистр
убираем из списка параметров session_key/access_token при наличии
**параметры сортируются лексикографически по ключам**
параметры соединяются в формате ключ=значение
sig = MD5(значения_параметров + session_secret_key)
значение sig переводится в нижний регистр

**Comment by kullfar** [comment]
I see.
Link to the docs is a good point.
I'll try to experiment with Exmaple to get this error

**Comment by kullfar** [comment]
btw, are this order requirement is for OAuth flow part and not for the API requests.
It can be ScribeJava out-of-scope thing to reorder params in API requests

**Comment by kullfar** [comment]
Ah. There is no need to change the actual order of params in the URL. It's needed to signature calculation only...
ook. seems PR is valuable ;-)

**Comment by aliksz** [comment]
Could you say, approximately, when pull request will be merged?

**Comment by kullfar** [comment]
this week

**Comment by kullfar** [comment]
have merged to the master with my improvements

**Comment by kullfar** [comment]
3.1.0 will be in central soon

**Pull request closed without merging**